### PR TITLE
Make DB logging API stricter; rename add_comparison -> db_comparison etc

### PR DESCRIPTION
**Pull request by peterjc**
I want to change the command line and internal API to make adding dependent table rows more configurable. In particular currently adding a comparison will implicitly add the configuration and genome rows if not already there - which may be undesirable (e.g. mismatch between the intended configuration already in the DB, and what snakemake is trying to use instead which if wrong will not already exist). This was deliberate while bootstrapping the framework as I could use the log-comparison functionality in isolation, creating a new DB as needed.

With these changes ``log_comparison`` and the ``log_<method>`` functions like ``log_fastani`` will now REQUIRE that the database exists and contains the configuration and genomes already.

This ensures our snakemake workflows WILL FAIL if they are trying to write to a non-existent configuration due to an error in propagating the configuration settings.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update

## Action Checklist

- [x] Work on a single issue/concept (if there are multiple separate issues to address, please use a separate pull request for each)
- [ ] Fork the `pyani-plus` repository under your own account (please [allow write access for repository maintainers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
- [x] Set up an appropriate development environment (please see `CONTRIBUTING.md` for this repository)
- [x] Create a new branch with a short, descriptive name
- [x] Work on this branch
  - [x] style guidelines have been followed (please see `CONTRIBUTING.md` for this repository)
  - [x] new code is commented, especially in hard-to-understand areas
  - [ ] corresponding changes to documentation have been made
  - [x] tests for the change have been added that demonstrate the fix or feature works
- [x] Test locally with `pytest-plus -v` **non-passing code will not be merged**
- [x] Rebase against `origin/master`
- [x] Check changes with linting/formatting tools before submission (please see `CONTRIBUTING.md` for this repository)
- [x] Commit branch
- [x] Submit pull request, describing the content and intent of the pull request
- [x] Request a code review
- [x] Continue the discussion at [`Pull requests` section](https://github.com/widdowquinn/pyani-plus/pulls) in the `pyan-plus` repository

### CODE REVIEW

File: pyani_plus/db_orm.py [hunk 1]
```diff
@@ -521,17 +521,37 @@ def add_configuration(  # noqa: PLR0913
     maxmatch: bool | None = None,
     kmersize: int | None = None,
     minmatch: float | None = None,
+    *,
+    create: float = False,
```
Comment (widdowquinn) [inline pyani_plus/db_orm.py:525]: `bool`?

File: pyani_plus/db_orm.py [hunk 2]
```diff
@@ -564,29 +587,42 @@ def add_configuration(  # noqa: PLR0913
     return config
 
 
-def add_genome(session: Session, fasta_filename: Path | str, md5: str) -> Genome:
+def db_genome(
+    session: Session, fasta_filename: Path | str, md5: str, *, create: float = False
```
Comment (widdowquinn) [inline pyani_plus/db_orm.py:591]: `create` a `bool`?

**Comment by peterjc** [comment]
We debated splitting these functions in two, perhaps ``get_configuration`` (return matching DB object if already present, error otherwise or perhaps return ``None``) and ``add_configuration`` (should this error if already there?) functions.

**Comment by peterjc** [comment]
Some conflicts from merging the ANIb support... will have to redo my conflict resolution.

**Pull request merged**